src/execution/execute.ts

```diff
@@ -261,6 +265,7 @@ export interface ExecutionArgs {
   fieldResolver?: Maybe<GraphQLFieldResolver<any, any>>;
   typeResolver?: Maybe<GraphQLTypeResolver<any, any>>;
   subscribeFieldResolver?: Maybe<GraphQLFieldResolver<any, any>>;
+  signal?: AbortSignal;
```

Comment:
ExecutionArgs is also the argument to `subscribe`/`experimentalSubscribeIncrementally`. So this PR allows you to specify `signal` to subscription APIs. I would expect that if I passed `signal` to a subscription API and then I aborted the signal at some point in the middle of a subscription (say, between source events), that it would also complete the subscription somehow, but I don't think your implementation does this.

Personally I think it would be reasonable for this PR to only work with non-subscription execution (and subscription support could be added later) — so my suggestion is for `experimentalSubscribeIncrementally` to throw if `signal` is provided, unless you're familiar with subscriptions and are motivated to make it work there too!

Reply:
Would be happy to try implementing it for Subscriptions but I think it makes sense to reduce this PR scope to minimal.

I added error throwing in case a client tries to pass an abort signal to subscriptions execution, thanks for the suggestion!